src/org/labkey/targetedms/TargetedMSController.java

```diff
@@ -3698,6 +3695,14 @@ protected QueryView createQueryView(RunDetailsForm form, BindException errors, b
             TargetedMSSchema schema = new TargetedMSSchema(getUser(), getContainer());
             return schema.createView(getViewContext(), settings, errors);
         }
+
+        @Override
+        protected ModelAndView getHtmlView(RunDetailsForm form, BindException errors) throws Exception
+        {
+            VBox vBox = (VBox) super.getHtmlView(form, errors);
+            vBox.addView(new InstrumentSummaryWebPart(form.getViewContext()), 1);
```

Comment:
We need to make this go through createQueryView() to create the instrument summary grid. Otherwise, when you export the grid it won't be filtered the same.

This line should call createQueryView() and pass in a name for the data region.

Then make the createQueryView() method check for the name and hand out the appropriate view. InstrumentSummaryWebPart is already a QueryView so it should be easy. If you pass in RunDetailsForm instead of the ViewContext, you'll have easy access to an already-parsed run id.

Reply:
converted to createQueryView, can not pass RunDetailsForm in the existing constructor as the InstrumentSummaryWebPart is being constructed as WebPartFactory.